sound/soc/sdw_utils/soc_sdw_utils.c

```diff
@@ -488,19 +488,25 @@ struct asoc_sdw_codec_info codec_info_list[] = {
 		.part_id = 0x3556,
 		.dais = {
 			{
-				.direction = {true, true},
+				.direction = {true, false},
 				.dai_name = "cs35l56-sdw1",
 				.dai_type = SOC_SDW_DAI_TYPE_AMP,
-				.dailink = {SOC_SDW_AMP_OUT_DAI_ID, SOC_SDW_AMP_IN_DAI_ID},
+				.dailink = {SOC_SDW_AMP_OUT_DAI_ID, SOC_SDW_UNUSED_DAI_ID},
 				.init = asoc_sdw_cs_amp_init,
 				.rtd_init = asoc_sdw_cs_spk_rtd_init,
 				.controls = generic_spk_controls,
 				.num_controls = ARRAY_SIZE(generic_spk_controls),
 				.widgets = generic_spk_widgets,
 				.num_widgets = ARRAY_SIZE(generic_spk_widgets),
 			},
+			{
+				.direction = {false, true},
+				.dai_name = "cs35l56-sdw1c",
```

Comment:
If only the capture DAI is connected (not going to happen?) then the asoc_sdw_cs_amp_init() is not going to be called, but it is true that the amp_init() only cares fro playback.
I'm not sure if something is missing or need to be taken care for this kind of use.

Likely nothing is needed, just checking.

Reply:
The only source of capture data is internal feedbacks from the playback stream. There isn't any external source, so the playback would have to be connected for the capture to be useful.